dtfs-central-api/src/v1/controllers/utilisation-report-service/get-utilisation-report-reconciliation-details-by-id.controller/helpers/filter-fee-record-payment-entity-groups.test.ts

```diff
@@ -0,0 +1,277 @@
+import { CURRENCY, FeeRecordEntityMockBuilder, PaymentEntityMockBuilder } from '@ukef/dtfs2-common';
+import { aUtilisationReport, getSqlIdGenerator } from '../../../../../../test-helpers';
+import { FeeRecordPaymentEntityGroup } from '../../../../../types/fee-record-payment-entity-group';
+import { filterFeeRecordPaymentEntityGroups } from './filter-fee-record-payment-entity-groups';
+import { ValidatedPaymentDetailsFilters } from '../../../../../types/utilisation-reports';
+
+describe('get-utilisation-report-reconciliation-details-by-id.controller helpers', () => {
+  const feeRecordIdGenerator = getSqlIdGenerator();
+
+  const aFeeRecord = () => FeeRecordEntityMockBuilder.forReport(aUtilisationReport()).build();
+
+  const aFeeRecordWithFacilityId = (facilityId: string) =>
+    FeeRecordEntityMockBuilder.forReport(aUtilisationReport()).withId(feeRecordIdGenerator.next().value).withFacilityId(facilityId).build();
+
+  describe('filterFeeRecordPaymentEntityGroups', () => {
+    describe('facility id filter', () => {
+      it('returns the group where the supplied facility id filter matches one of the fee records in a group', () => {
+        // Arrange
+        const firstGroup: FeeRecordPaymentEntityGroup = {
+          feeRecords: [aFeeRecordWithFacilityId('11111111'), aFeeRecordWithFacilityId('22222222')],
+          payments: [PaymentEntityMockBuilder.forCurrency(CURRENCY.GBP).build()],
+        };
+
+        const secondGroup: FeeRecordPaymentEntityGroup = {
+          feeRecords: [aFeeRecordWithFacilityId('33333333')],
+          payments: [PaymentEntityMockBuilder.forCurrency(CURRENCY.GBP).build()],
+        };
+
+        const groups = [firstGroup, secondGroup];
+
+        const facilityIdFilter = '22222222';
+        const filters = { facilityId: facilityIdFilter };
+
+        // Act
+        const result = filterFeeRecordPaymentEntityGroups(groups, filters);
+
+        // Assert
+        expect(result).toContainEqual(firstGroup);
```

Comment:
is it possible to check the full array coming back?  should the array only be `[firstGroup]`?

Reply:
Changed these tests and the existing ones in this file to match that pattern